src/redux/connection/connectionMiddleware.ts

```diff
@@ -1,40 +1,48 @@
+import { createAction, createAsyncThunk } from '@reduxjs/toolkit';
+
 import { api } from '@/api/api';
 import type { AppStartListening } from '@/redux/listenerMiddleware';
-import { initializeLuaApi } from '@/redux/luaapi/luaApiSlice';
 
 import { onCloseConnection, onOpenConnection, startConnection } from './connectionSlice';
 
-let openspace: OpenSpace.openspace;
+export const closeConnection = createAction<void>('closeConnection');
+
+export const connectToOpenSpace = createAsyncThunk(
+  'connection/connectToOpenSpace',
+  async (_, thunkAPI) => {
+    async function onConnect() {
+      thunkAPI.dispatch(onOpenConnection());
+    }
+
+    function onDisconnect() {
+      thunkAPI.dispatch(onCloseConnection());
+
+      let reconnectionInterval = 1000;
+      setTimeout(() => {
+        api.connect();
+        reconnectionInterval += 1000;
```

Comment:
Interesting, didn't know that we increased the interval with a second at every attempt. Wonder if this is still meaningful?

Reply:
I think this is a way to basically not create an interval. It will just make retries with 1 second interval